File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        for (int i = 0; i < nums.size(); i++) {
            int cmpl = target - nums[i];
            if (numMap.count(cmpl)) {
                return {numMap[cmpl], i};
            }
            numMap[nums[i]] = i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numToIndex = {}
        for i, currNum in enumerate(nums):
            complement = target - currNum
            if complement in numToIndex:
                return [numToIndex[complement], i]
            numToIndex[currNum] = i
        return []